presto-main/src/main/java/com/facebook/presto/sql/gen/LambdaBytecodeGenerator.java

```diff
@@ -197,27 +196,7 @@ public static BytecodeNode generateLambda(
         }
 
         if (MethodHandle.class.equals(lambdaInterface)) {
-            // generate captured lambda expression as MethodHandle
-            List<BytecodeExpression> bytecodeCaptureVariables = captureVariableBuilder.build().stream()
-                    .map(variable -> variable.cast(Object.class))
-                    .collect(toImmutableList());
-
-            Variable functionVariable = scope.createTempVariable(MethodHandle.class);
-            block.append(context.generate(lambda, lambdaInterface));
-            block.append(
-                    new IfStatement()
-                            .condition(wasNull)
-                            // ifTrue: do nothing i.e. Leave the null MethodHandle on the stack, and leave the wasNull variable set to true
-                            .ifFalse(
-                                    new BytecodeBlock()
-                                            .putVariable(functionVariable)
-                                            .append(invokeStatic(
-                                                    MethodHandles.class,
-                                                    "insertArguments",
-                                                    MethodHandle.class,
-                                                    functionVariable,
-                                                    constantInt(0),
-                                                    newArray(type(Object[].class), bytecodeCaptureVariables)))));
+            throw new PrestoException(FUNCTION_IMPLEMENTATION_ERROR, "Lambda cannot be generated as MethodHandle");
```

Comment:
If this is impossible, throw a different exception (like VerifyException)

In the exception (or comments), make it explicit that this is should not happen due to prior sanity checks.